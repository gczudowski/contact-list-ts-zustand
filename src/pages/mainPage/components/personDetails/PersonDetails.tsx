import React, { ReactElement } from 'react';
import { createUseStyles } from 'react-jss';

type Props = {
  firstNameLastName: string;
  jobTitle: string;
};

function PersonDetails({ firstNameLastName, jobTitle }: Props): ReactElement {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <div className={classes.firstNameLastNameLabel}>{firstNameLastName}</div>
      <div className={classes.jobTitleLabel}>{jobTitle.toUpperCase()}</div>
    </div>
  );
}

const useStyles = createUseStyles({
  container: {
    display: 'flex',
    flexDirection: 'column',
  },
  firstNameLastNameLabel: {
    paddingBottom: 0,
    color: '#333333',
    fontSize: '20px',
    fontWeight: 700,
  },
  jobTitleLabel: {
    color: '#e74c3c',
    fontSize: '13px',
    fontWeight: 400,
  },
});

export default PersonDetails;
