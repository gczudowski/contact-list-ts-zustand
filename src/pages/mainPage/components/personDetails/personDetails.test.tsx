import React from 'react';
import renderer from 'react-test-renderer';
import PersonDetails from './PersonDetails';

describe('PersonDetails', () => {
  const firstNameLastNameMock = 'firstNameLastNameMock';
  const jobTitleMock = 'jobTitleMock';

  it('renders properly with initial props', () => {
    const component = renderer.create(
      <PersonDetails firstNameLastName={firstNameLastNameMock} jobTitle={jobTitleMock} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
