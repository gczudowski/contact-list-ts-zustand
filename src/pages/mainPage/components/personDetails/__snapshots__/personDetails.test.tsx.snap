// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDetails renders properly with initial props 1`] = `
<div
  className="container-0-2-1"
>
  <div
    className="firstNameLastNameLabel-0-2-2"
  >
    firstNameLastNameMock
  </div>
  <div
    className="jobTitleLabel-0-2-3"
  >
    JOBTITLEMOCK
  </div>
</div>
`;
